android/src/main/java/io/invertase/googlemobileads/ReactNativeGoogleMobileAdsBannerAdViewManager.java

```diff
@@ -98,62 +111,62 @@ public void setRequest(ReactViewGroup reactViewGroup, ReadableMap value) {
     requestAd(reactViewGroup);
   }
 
-  @ReactProp(name = "size")
-  public void setSize(ReactViewGroup reactViewGroup, String value) {
-    size = ReactNativeGoogleMobileAdsCommon.getAdSize(value, reactViewGroup);
-
-    WritableMap payload = Arguments.createMap();
-
-    int width = size.getWidth();
-    int height = size.getHeight();
-
-    payload.putDouble("width", width);
-    payload.putDouble("height", height);
-
-    if (size != AdSize.FLUID) {
-      sendEvent(reactViewGroup, "onSizeChange", payload);
+  @ReactProp(name = "sizes")
+  public void setSizes(ReactViewGroup reactViewGroup, ReadableArray value) {
+    List<AdSize> sizeList = new ArrayList<>();
+    for (Object size : value.toArrayList()) {
+      if (size instanceof String) {
+        String sizeString = (String) size;
+        sizeList.add(ReactNativeGoogleMobileAdsCommon.getAdSize(sizeString, reactViewGroup));
+      }
     }
+    sizes = sizeList;
     requestAd(reactViewGroup);
   }
 
-  private void setAdListener(ReactViewGroup reactViewGroup) {
-    final AdView adView = getAdView(reactViewGroup);
+  @ReactProp(name = "manualImpressionsEnabled")
+  public void setManualImpressionsEnabled(ReactViewGroup reactViewGroup, boolean value) {
+    this.manualImpressionsEnabled = value;
+    requestAd(reactViewGroup);
+  }
 
+  private BaseAdView initAdView(ReactViewGroup reactViewGroup) {
+    BaseAdView oldAdView = getAdView(reactViewGroup);
+    if (oldAdView != null) {
+      oldAdView.destroy();
+      reactViewGroup.removeView(oldAdView);
+    }
+    BaseAdView adView;
+    if (ReactNativeGoogleMobileAdsCommon.isAdManagerUnit(unitId)) {
+      adView = new AdManagerAdView(reactViewGroup.getContext());
+    } else {
+      adView = new AdView(reactViewGroup.getContext());
+    }
     adView.setDescendantFocusability(ViewGroup.FOCUS_BLOCK_DESCENDANTS);
     adView.setAdListener(
         new AdListener() {
           @Override
           public void onAdLoaded() {
-            int top;
-            int left;
-            int width;
-            int height;
-
-            if (size == AdSize.FLUID) {
-              top = 0;
+            AdSize adSize = adView.getAdSize();
+            int left, top, width, height;
+            if (isFluid) {
+              // TODO size=FLUID is still not working
```

Comment:
Extend AdSize support to handle size=FLUID

I'm fine with merging the main Google Ad Manager PR - but we should peel this off as a new issue for enhancement tracking I think